plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/mutate/MutateProcessor.java

```diff
@@ -0,0 +1,312 @@
+
+import org.elasticsearch.ingest.Data;
+import org.elasticsearch.ingest.processor.Processor;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public final class MutateProcessor implements Processor {
+
+    public static final String TYPE = "mutate";
+
+    private final Map<String, Object> update;
+    private final Map<String, String> rename;
+    private final Map<String, String> convert;
+    private final Map<String, String> split;
+    private final Map<String, List<String>> gsub;
+    private final Map<String, String> join;
+    private final List<String> remove;
+    private final List<String> strip;
+    private final List<String> uppercase;
+    private final List<String> lowercase;
+
+    public MutateProcessor(Map<String, Object> update,
+                           Map<String, String> rename,
+                           Map<String, String> convert,
+                           Map<String, String> split,
+                           Map<String, List<String>> gsub,
+                           Map<String, String> join,
+                           List<String> remove,
+                           List<String> strip,
+                           List<String> uppercase,
+                           List<String> lowercase) {
+        this.update = update;
+        this.rename = rename;
+        this.convert = convert;
+        this.split = split;
+        this.gsub = gsub;
+        this.join = join;
+        this.remove = remove;
+        this.strip = strip;
+        this.uppercase = uppercase;
+        this.lowercase = lowercase;
+    }
+
+    @Override
+    public void execute(Data data) {
+        doUpdate(data);
+        doRename(data);
+        doConvert(data);
+        doSplit(data);
+        doGsub(data);
+        doJoin(data);
+        doRemove(data);
+        doStrip(data);
+        doUppercase(data);
+        doLowercase(data);
+    }
+
+    private void doUpdate(Data data) {
+        if (update == null) { return; }
+
+        for(Map.Entry<String, Object> entry : update.entrySet()) {
+            data.addField(entry.getKey(), entry.getValue());
+        }
+    }
+
+    private void doRename(Data data) {
+        if (rename == null) { return; }
+
+        for(Map.Entry<String, String> entry : rename.entrySet()) {
+            Object oldVal = data.getProperty(entry.getKey());
+            data.addField(entry.getValue(), oldVal);
+        }
+    }
+
+    private Object parseValueAsType(Object oldVal, String toType) {
+        if("integer".equals(toType)) {
+            oldVal = Integer.parseInt(oldVal.toString());
+        } else if("float".equals(toType)) {
+            oldVal = Float.parseFloat(oldVal.toString());
+        } else if("string".equals(toType)) {
+            oldVal = oldVal.toString();
+        } else if("boolean".equals(toType)) {
+            String val = oldVal.toString();
+            if (val.equals("true") || val.equals("t") || val.equals("yes") || val.equals("y") || val.equals("1")) {
```

Comment:
but since we can't rely directly on ES, we maybe need to add an exception here... and just allow it for now until this becomes an external library. I don't like to duplicate logic and tests.